### PR TITLE
Add custom auth URLs and simplify `get`/etc methods

### PR DESCRIPTION
**Pull request by marioortizmanero**
## Description

This adds a configurable prefix URL for authentication.

I've also simplified the request methods in the base client. We don't really need `get` if it's only used in `endpoint_get`. We can also extend `post_form` to `auth_post`, which also prepends the auth URL for convenience, just like `endpoint_get`. Finally, I've renamed `endpoint_get` and similars to `api_get` for consistency and because it's clearer and much shorter.

## Motivation and Context

See #350 

## Dependencies 

None

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?

I've modified the `endpoint_url` test, and added another one for `auth_url`.

## Is this change properly documented?

TODO

### CODE REVIEW

File: doc/uml/trait_hierarchy.plantuml [hunk 1]
```diff
@@ -10,6 +10,7 @@ abstract class BaseClient {
     ClientResult<()> auto_reauth()
     ClientResult<()> refresh_token()
     String endpoint_url()
+    String auth_url()
```
Comment (marioortizmanero) [inline doc/uml/trait_hierarchy.plantuml:13]: @ramsayleung, can you generate the diagram again? Do you know if there's a way to do this automatically instead?

**Comment by marioortizmanero** [comment]
Note that #375 should be merged before this.

File: doc/uml/trait_hierarchy.plantuml (hunk 1 above)
Comment (ramsayleung) [reply doc/uml/trait_hierarchy.plantuml:13]: Yes, I can regenerate this diagram, but generating the diagram manually is a little clumsy, I would like to figure out how to generate this diagram automatically.

File: src/clients/mod.rs [hunk 2]
```diff
@@ -55,19 +55,58 @@ mod test {
     }
 
     #[test]
-    fn test_endpoint_url() {
-        let spotify = ClientCredsSpotify::default();
+    fn test_api_url() {
+        let mut spotify = ClientCredsSpotify::default();
```
Comment (ramsayleung) [inline src/clients/mod.rs:59]: Is it necessary to add the `mut` identifier to mark `spotify` mutable? It looks like the `api_url` doesn't require a mutable `Spotify`.

File: doc/uml/trait_hierarchy.plantuml (hunk 1 above)
Comment (ramsayleung) [reply doc/uml/trait_hierarchy.plantuml:13]: I think we could use this GitHub Action to generate an image from `plantuml` file automatically: https://github.com/marketplace/actions/generate-plantuml

**Comment by ramsayleung** [comment]
Everything looks good to me, except for the discussion about how to generate an image file from plantuml file automatically.

File: doc/uml/trait_hierarchy.plantuml (hunk 1 above)
Comment (marioortizmanero) [reply doc/uml/trait_hierarchy.plantuml:13]: Great! Then you can ignore this comment, and we'll set it up in a separate PR.

**Review by ramsayleung — APPROVED** [review]

**Pull request merged**